src/iter/noop.rs

```diff
@@ -33,6 +33,11 @@ impl<T> Folder<T> for NoopConsumer {
         self
     }
 
+   fn consume_iter<I>(self, iter: I) -> Self where I: IntoIterator<Item=T> {
+        for _ in iter {}
```

Comment:
Man, seeing this silliness really makes me revisit the [question of side-effects](https://github.com/rayon-rs/rayon/issues/352). It'd be so nice to just "do nothing" here. In any case, I think I would prefer `iter.for_each(|_| ())`.

Reply:
Ditto on using `fold` instead.